File: app/src/main/java/com/vikingsen/cheesedemo/ux/cheeselist/CheeseListContract.java
package com.vikingsen.cheesedemo.ux.cheeselist;


import com.vikingsen.cheesedemo.model.database.cheese.Cheese;

import java.util.List;

interface CheeseListContract {
    interface View {
        void showLoading(boolean loading);
        void showCheeses(List<Cheese> cheeses);
        void showError();
    }
}

File: app/src/main/java/com/vikingsen/cheesedemo/ux/cheeselist/CheeseListContract.kt
package com.vikingsen.cheesedemo.ux.cheeselist


import com.vikingsen.cheesedemo.model.room.cheese.Cheese

interface CheeseListContract {
    interface View {
        fun showLoading(loading: Boolean)
        fun showCheeses(cheeses: List<Cheese>)
        fun showError()
    }
}
